extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -594,3 +633,59 @@ export async function promptSelectDatabase(connectionInfo: IConnectionInfo): Pro
 	}
 	return selectedDatabase;
 }
+
+export async function getConnectionURI(connectionInfo: IConnectionInfo): Promise<string | undefined> {
+	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+
+	let connectionURI: string = '';
+	try {
+		await vscode.window.withProgress(
+			{
+				location: vscode.ProgressLocation.Notification,
+				title: constants.connectionProgressTitle,
+				cancellable: false
+			}, async (_progress, _token) => {
+				// show progress bar while connecting to the users selected connection profile
+				connectionURI = await vscodeMssqlApi.connect(connectionInfo!);
+			}
+		);
+	} catch (e) {
+		// mssql connection error will be shown to the user
+		return undefined;
+	}
+
+	return connectionURI;
+}
+
+export async function promptSelectTable(connectionURI: string, selectedDatabase: string, bindingType: BindingType): Promise<string | undefined> {
+	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+	const userObjectName = bindingType === BindingType.input ? constants.enterObjectName : constants.enterObjectNameToUpsert;
+
+	// Create query to get list of tables from database selected
+	let tableQuery = tablesQuery(selectedDatabase);
+	const params = { ownerUri: connectionURI, queryString: tableQuery };
+	let queryResult: azureFunctionsContracts.SimpleExecuteResult = await vscodeMssqlApi.sendRequest(azureFunctionsContracts.SimpleExecuteRequest.type, params);
+
+	// Get table names from query result rows
+	const tableNames = queryResult.rows.map(r => r[0].displayValue);
+	tableNames.push(userObjectName);
+	// prompt user to select table from list of tables
+	let selectedObject = await vscode.window.showQuickPick(tableNames, {
+		canPickMany: false,
+		title: constants.selectTable,
+		ignoreFocusOut: true
+	});
+
+	if (selectedObject === constants.enterObjectName) {
+		// user manually enters table or view to query or upsert into
+		return vscode.window.showInputBox({
+			prompt: bindingType === BindingType.input ? constants.sqlTableOrViewToQuery : constants.sqlTableToUpsert,
+			placeHolder: constants.placeHolderObject,
+			validateInput: input => input ? undefined : constants.nameMustNotBeEmpty,
+			ignoreFocusOut: true
+		}) ?? '';
+	}
+	return selectedObject;
+}
+
+export function tablesQuery(selectedDatabase: string): string { return `select concat('[',table_schema,']', '.', '[',table_name,']') from ${selectedDatabase}.INFORMATION_SCHEMA.TABLES where TABLE_TYPE = 'BASE TABLE'`; }
```

Comment:
You aren't escaping the names so this is incorrect. You should use the QUOTENAME T-SQL function